curvefs/src/tools/list/curvefs_topology_list.h

```diff
@@ -0,0 +1,169 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+/*
+ * Project: curve
+ * Created Date: 2021-12-17
+ * Author: chengyi01
+ */
+#ifndef CURVEFS_SRC_TOOLS_LIST_CURVEFS_TOPOLOGY_LIST_H_
+#define CURVEFS_SRC_TOOLS_LIST_CURVEFS_TOPOLOGY_LIST_H_
+
+#include <brpc/channel.h>
+#include <gflags/gflags.h>
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "curvefs/proto/topology.pb.h"
+#include "curvefs/src/mds/common/mds_define.h"
+#include "curvefs/src/tools/curvefs_tool.h"
+#include "curvefs/src/tools/curvefs_tool_define.h"
+
+namespace curvefs {
+namespace tools {
+namespace list {
+
+/**
+ * @brief
+ * pool redundanceandplacementpolicy
+ *
+ * @details
+ */
+struct PoolPolicy {
+    uint32_t copysetNum;
+    uint32_t replicaNum;
+    uint32_t zoneNum;
+    bool error = false;
+
+    explicit PoolPolicy(const std::string& jsonStr);
+
+    friend std::ostream& operator<<(std::ostream& os, const PoolPolicy& policy);
+};
+
+// TODO(chengyi01): output a json file which can be used to build topology
+class TopologyListTool
+    : public CurvefsToolRpc<curvefs::mds::topology::ListTopologyRequest,
+                            curvefs::mds::topology::ListTopologyResponse,
+                            curvefs::mds::topology::TopologyService_Stub> {
+ public:
+    explicit TopologyListTool(const std::string& cmd = kTopologyListCmd,
+                              bool show = true)
+        : CurvefsToolRpc(cmd) {
+        show_ = show;
+    }
+    void PrintHelp() override;
+    int Init() override;
+
+ protected:
+    void AddUpdateFlags() override;
+    bool AfterSendRequestToHost(const std::string& host) override;
+
+    /**
+     * @brief Get the PoolInfo From Response, fill into poolId2PoolInfo
+     * not include zoneId list (will be filled in GetZoneInfoFromResponse)
+     *
+     * @return true
+     * @return false
+     * @details
+     *
+     */
+    bool GetPoolInfoFromResponse();
+
+    /**
+     * @brief Get the ZoneInfo From Response, fill into zoneId2ZoneInfo
+     * not include serverId list (will be filled in GetServerInfoFromResponse)
+     * will fill poolId2PoolInfo's zoneId list
+     *
+     * @return true
+     * @return false
+     * @details
+     */
+    bool GetZoneInfoFromResponse();
+
+    /**
+     * @brief Get the ServerInfo From Response, fill into serverId2ServerInfo
+     * not include serverId list
+     * will fill zoneId2ZoneInfo's serverId list
+     *
+     * @return true
+     * @return false
+     * @details
+     */
+    bool GetServerInfoFromResponse();
+
+    /**
+     * @brief Get the MetaserverInfo From Response,
+     * fill into metaserverId2MetaserverInfo
+     * will fill serverId2ServerInfo's metaserverId list
+     *
+     * @return true
+     * @return false
+     * @details
+     */
+    bool GetMetaserverInfoFromResponse();
+
+ protected:
+    std::string clusterId_;
```

Comment:
Maybe separate the variables and functions is more clearer.

Reply:
> Maybe separate the variables and functions is more clearer.

fix